examples/ReadWriteSplittingSample/src/main/java/software/amazon/ReadWriteSplittingSample.java

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package software.amazon;
+
+import com.zaxxer.hikari.HikariConfig;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Properties;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+import software.amazon.jdbc.ConnectionProviderManager;
+import software.amazon.jdbc.HikariPooledConnectionProvider;
+import software.amazon.jdbc.HostSpec;
+import software.amazon.jdbc.PropertyDefinition;
+import software.amazon.jdbc.plugin.readwritesplitting.ReadWriteSplittingPlugin;
+
+public class ReadWriteSplittingSample {
+  // Adjust this value to run the sample with different driver configurations:
+  // 1: no r/w splitting
+  // 2: r/w splitting using r/w plugin
+  // 3: r/w plugin with internal connection pools
+  static final int APPROACH_ID = 1;
```

Comment:
i think we should abandon the approach idea and have one sample file for each approach

Reply:
That would simplify the file, but there would also be more code duplication and users would no longer be able to view all the code at once in a single file. Do we still want to go ahead with splitting this into different files?

Reply:
Multiple files follow some arbitrary guidelines that I have seen since working on the wrapper. We tend to have multiple small files that handle one single case so users can run/copy them. If you believe the "approach" strategy matches better the way people will read it and the blog post, that's completely fine by me too.

Maybe once the blog post is released we can think whether we need smaller snippets for people to copy